types/mempool/priority_nonce.go

```diff
@@ -226,6 +226,9 @@ func (mp *PriorityNonceMempool[C]) Insert(ctx context.Context, tx sdk.Tx) error
 
 	// if it's an unordered tx, we use the timeout timestamp instead of the nonce
 	if unordered, ok := tx.(sdk.TxWithUnordered); ok && unordered.GetUnordered() {
+		if nonce > 0 {
```

Comment:
```suggestion
		if sequenceNonce > 0 {
```
we should rename this variable for clarity. sequence and timestamp are both noncces

Reply:
i've unified nonce selection code in this commit [unify nonce selection code](https://github.com/cosmos/cosmos-sdk/pull/24581/commits/5123411b3af5981a8e81fdfeddc9e8f2e8031762), since it is copy pasted 4 times. lmk if this works.